### DISCUSSION TITLE
Support for Biome

### DISCUSSION THREAD

**Discussion by vaquilina-work** [TanStack/query#7525] [Ideas]
Thank you for including an eslint plugin along with react query. We have recently migrated to [biome](https://biomejs.dev/). Is there a possibility of a plugin for biome in our future?

**Comment by justin13888** [comment]
Sounds interesting to me as well because some of my projects have also migrated to biome recently. The Eslint plugin for React query adds a number of good rules and to do the same on Biome would require a plugin of some sort. However, it seems a spec for Biome plugins is still being actively discussed so it may not be feasible?? Welcome thoughts from anyone who may know more about Biome

Biome plugins proposal: <https://github.com/biomejs/biome/discussions/1762>

**Comment by qcho** [comment]
Biome v2 now supports plugins officially. It uses gritQL language to create rules quite easily.

https://biomejs.dev/linter/plugins/

**Comment by suryaashish-tyke** [comment]
Biome now supports plugins officially, any update on this? and also one for Tanstack Router

**Comment by davejsdev** [comment]
Really hope this comes soon! I'm setting up a new project now for work and I'm using TanStack Start + Biome.

I'm probably still going to plow ahead with the expectation that this will come eventually, just sucks that the DX is going to suffer until then. 🫠

Nonetheless, so grateful we have such high quality open-source tools and appreciate everyone who works on them!

**Reply by qcho** [reply]
I'm using it too. Works like a charm! remember to activate domains ^_^

```json
    "domains": {
      "project": "none", // it's slow
      "react": "recommended",
      "test": "recommended"
    }
```

Would be nice to have a tanstack domain too ^_^

**Comment by uncvrd** [comment]
I came across this while looking for tanstack router biome support for "sorted keys" that Tanstack Router has for eslint. It would be real nice to have a `biome` assist for `useSortedKeys` since I use that for the rest of my project but all 200+ of my router files need a.

```
// biome-ignore assist/source/useSortedKeys: ordering required for type inference
export const Route = createFileRoute("/(app)/app/_layout/workspace/campaigns")({
``` 

since the keys need a specific order for type inference and should not be modified...

I am sure it would be useful for `useInfiniteQuery` as I believe those need to be sorted in a specific order as well if I remember correctly

**Comment by marcopedone** [comment]
A biome plugin would be helpful 👍

**Comment by jchook** [comment]
Biome's plugin system is not powerful enough to express all of the rules in `eslint-plugin-query`.
